pdns/dnsdistdist/dnsdist-tcp-downstream.cc

```diff
@@ -257,9 +258,11 @@ static void prepareQueryForSending(TCPQuery& query, uint16_t queryID, Connection
   editPayloadID(query.d_buffer, queryID, query.d_proxyProtocolPayloadAdded ? query.d_idstate.d_proxyProtocolPayloadSize : 0, true);
 }
 
+static const string classnamePrefix = "TCPConnectionToBackend::";
 IOState TCPConnectionToBackend::queueNextQuery(std::shared_ptr<TCPConnectionToBackend>& conn)
 {
   conn->d_currentQuery = std::move(conn->d_pendingQueries.front());
+  auto closer = conn->d_currentQuery.d_query.d_idstate.getCloser(classnamePrefix + __func__);
```

Comment:
I'm pretty sure you tried and there was an issue, but would `__PRETTY_FUNCTION__` work? It would save a few allocations if it does.